### PR TITLE
fix: I should be able to use "enter" key to create profile

### LINKED ISSUE #4808: I should be able to use "enter" key to create profile
Enter key doesn't seem to work to move to the next stage here:

Screenshot 2024-04-04 at 15 09 19

### PR DESCRIPTION
**Pull request by ijreilly**
## Context
Fixes #4808 

TL;DR
Introducing pure stateless modal component ("UI modal") for our auth modal not to have default hotkeyScope overriding our create-profile hotkeyScope

Explanation
create-profile hotkey scope is set by PageChangeEffect; CreateProfile component adds enter key shortcut; but this scope is overwritten by the default scope by the Modal component that expects a hotkeyScope to reset to (and defaults to the default hotkeyScope if none indicated). 
In the auth flow we were using that Modal component to give a modal look to the flow but it is not a modal per say, it's a set of pages contained within a modal look.
By creating this UI component we are escaping that hotkeyScope overriding that does not make sense in our context. 

## How was it tested
Locally
Storybook

### CODE REVIEW

File: packages/twenty-front/src/modules/auth/sign-in-up/hooks/useSignInUp.tsx [hunk 1]
```diff
@@ -118,31 +116,6 @@ export const useSignInUp = (form: UseFormReturn<Form>) => {
     ],
   );
 
-  useScopedHotkeys(
```
Comment (ijreilly) [inline packages/twenty-front/src/modules/auth/sign-in-up/hooks/useSignInUp.tsx:121]: Before the fix this was also overwritten by the Modal default hotkeyScope, but the behaviour was ensured by the handleKeyDow logic in SignInUpForm.

With the fix this works too while the logic is slightly different for the SignInUpStep.Password step, making them concurrent. We are replacing handleKeyDown logic with this one (adapting it with the desired logic, the one that has been executed until then, hence the move to SignInUpForm) as we want to favorise hotkeyScope.

**Review by charlesBochet — APPROVED** [review]

File: packages/twenty-front/src/pages/auth/CreateProfile.tsx [hunk 2]
```diff
@@ -126,19 +123,17 @@ export const CreateProfile = () => {
...
-    },
-    PageHotkeyScope.CreateProfile,
-    [onSubmit],
-  );
-
   if (onboardingStatus !== OnboardingStatus.OngoingProfileCreation) {
     return null;
   }
 
+  const onKeyDown = (event: React.KeyboardEvent<HTMLInputElement>) => {
```
Comment (charlesBochet) [inline packages/twenty-front/src/pages/auth/CreateProfile.tsx:130]: onNameInputKeyDown

**Review by charlesBochet — APPROVED** [review]
